mmtk/src/active_plan.rs

```diff
@@ -5,7 +5,34 @@ use mmtk::util::opaque_pointer::*;
 use mmtk::vm::ActivePlan;
 use mmtk::Mutator;
 use mmtk::Plan;
-use std::sync::Mutex;
+use std::sync::{Mutex, MutexGuard};
+
+struct OpenJDKMutatorIterator<'a> {
+    _guard: MutexGuard<'a, ()>,
+}
+
+impl<'a> OpenJDKMutatorIterator<'a> {
+    fn new(guard: MutexGuard<'a, ()>) -> Self {
+        // Reset mutator iterator
+        unsafe {
+            ((*UPCALLS).reset_mutator_iterator)();
```

Comment:
This line shows that an `OpenJDKMutatorIterator` instance is not really an independent instance.  All `OpenJDKMutatorIterator` instances are using the same global instance.